OpenImis.ePayment/Escape/Payment/Data/GepgUtility.cs

```diff
@@ -206,245 +187,157 @@ public String CreateGePGCancelPaymentRequest(IConfiguration Configuration, int P
-            XmlWriterSettings settings = null;
-            String outString = String.Empty;
+            var xs = new XmlSerializer(typeof(gepgBillCanclReq));
+            var ns = new XmlSerializerNamespaces();
+            ns.Add("", "");
+            var settings = new XmlWriterSettings() { OmitXmlDeclaration = true };
+            var sb = new StringBuilder();
 
-            XmlWriter xw = null;
-
-            try
+            using (var xw = XmlWriter.Create(sb, settings))
             {
-                settings = new XmlWriterSettings()
+                try
                 {
-                    OmitXmlDeclaration = true
-                };
-
-                ns = new XmlSerializerNamespaces();
-                ns.Add("", "");
-
-                StringBuilder sb = new StringBuilder();
-                xs = new XmlSerializer(typeof(gepgBillCanclReq));
-
-                xw = XmlWriter.Create(sb, settings);
+                    xs.Serialize(xw, gepgBillCanclReq, ns);
+                    xw.Flush();
 
-                xs.Serialize(xw, gepgBillCanclReq, ns);
-                xw.Flush();
+                    var outString = sb.ToString();
 
-                outString = sb.ToString();
-
-                var signature = this.GenerateSignature(outString);
-
-                GePGPaymentCancelRequest GePGPaymentCancelRequest = new GePGPaymentCancelRequest()
-                {
-                    gepgBillCanclReq = gepgBillCanclReq,
-                    gepgSignature = signature
-                };
+                    var signature = this.GenerateSignature(outString);
 
-                settings = new XmlWriterSettings();
-                sb = new StringBuilder();
-                xs = new XmlSerializer(typeof(GePGPaymentCancelRequest));
-                xw = XmlWriter.Create(sb, settings);
+                    GePGPaymentCancelRequest GePGPaymentCancelRequest = new GePGPaymentCancelRequest()
+                    {
+                        gepgBillCanclReq = gepgBillCanclReq,
+                        gepgSignature = signature
+                    };
 
-                xs.Serialize(xw, GePGPaymentCancelRequest, ns);
-                xw.Flush();
-                outString = sb.ToString();
+                    xs = new XmlSerializer(typeof(GePGPaymentCancelRequest));
+                    settings = new XmlWriterSettings();
+                    sb = new StringBuilder();
 
-            }
-            catch (Exception ex)
-            {
-                Console.WriteLine(ex.ToString());
-            }
-            finally
-            {
-                if (xw != null)
+                    using (var xw2 = XmlWriter.Create(sb, settings))
+                    {
+                        xs.Serialize(xw, gepgBillCanclReq, ns);
+                        xw.Flush();
+                    }
+                }
+                catch (Exception ex)
                 {
-                    xw.Close();
+                    Console.WriteLine(ex.ToString());
```

Comment:
Instead of logging the error in Console, can we log it to Log4Net or somewhere else to review the exception later?

Reply:
The same as above, i didn't want to modify the component to allow logging